firmware/isp.c

```diff
@@ -124,21 +125,31 @@ void ispConnect() {
 
 void ispDisconnect() {
 
+	cli();
 	/* set all ISP pins inputs */
 	ISP_DDR &= ~((1 << ISP_RST) | (1 << ISP_SCK) | (1 << ISP_MOSI));
 	/* switch pullups off */
 	ISP_OUT &= ~((1 << ISP_RST) | (1 << ISP_SCK) | (1 << ISP_MOSI));
+	sei();
 
 	/* disable hardware SPI */
 	spiHWdisable();
 }
 
 void ispSafeResetPulse(void)
 {
+	/* 
```

Comment:
I totally agree with the change, it could indeed be the cause of the usb issues. Nice catch!
Though I don't understand the comment "AVR cannot set multiple bits atomically" shouldn't this be "AVR cannot atomically read-modify-write a register" (cfr the assembler listing: an irq could come in between the in and out instruction). The fact that there are two bits (MOSI and SCK) is not a problem. Or am I missing something?